### PR TITLE
feat : Implemented Calculator app, added 'pwd' command, open all apps from terminal command, fixed Responsive design

### PR DESCRIPTION
**Pull request by sunitapt**
### 1. Added Terminal-Calculator app (#44)
--------
-   **Evaluate mathematical expressions**:

        1920/16*9
        2^32 
        12/5*9+9.4*2  
        19*(45%5)*((2/3)-(2/3))
        sqrt(1.4^2 + 1.5^2)  
       
    -   **Operators**: addition (`+`), subtraction (`-`), multiplication (`*`), division (`/`), modulo (`%`)exponentiation (`^`). 
    -   **Mathematical functions**: `abs`, `acos`, `acosh`, `asin`, `asinh`, `atan`, `atan2`, `atanh`, `ceil`, `cos`, `cosh`, `exp`,         `floor`, `ln`, `log`,`log10`, `log2`,`round`, `sin`, `sinh`,`sqrt`,`cbrt`,`tan`, `tanh`, `max`, `min`, `fac`, `random`,`pow`.

    -   **High-precision numeric type** with *20* significant digits.

    -   **Exponential notation**: `3.4142787736421956e+29`.
    -   **Variable assignments and use in futher evaluation** : `x=1` `x+2` 
    - **`clear` command for clearing  calculator app**
    -  **`exit` command for exit  from calculator app**      

### 2. Added `pwd` command for terminal and other commands to open all apps from the terminal like (#24 )
--------
- `spotify` &  `spotify .`   to open spotify app, `chrome` &  `chrome  .`   to open google chrome app,  `todoist` &  `todoist  .`   to open todo-ist app,  `trash` &  `trash  .`   to open trash app,  `settings` &  `settings  .`   to open settings app,  `sendmsg` &  `sendmsg .`   to open sendmsg , `about-vivek` &  `about-vivek  .`   to open about-vivek folder.

### 3.  Did responsive design fix for the all-application screen  (#39)
--------

- **Added app slicing for titles of apps**
- **Before (Laptop Screen) :**
Screenshot from 2021-08-31 14-55-17
- **After (Laptop Screen) :**
Screenshot from 2021-08-31 14-55-35
- **Before (Mobile Screen) :** 
Screenshot_20210831-145655
- **After (Mobile Screen) :**
Screenshot_20210831-145613

### CODE REVIEW

**Comment by vivek9patel** [comment]
Great work there, especially in calculator app!
I have few small changes.

File: src/components/apps/terminal.js [hunk 1]
```diff
@@ -237,16 +237,76 @@ export class Terminal extends Component {
...
+                let str = this.current_directory;  
+                result = str.replace("~","/home/vivek")
+                break;
             case "code":
-                if(words[0]==="."){
+                if(words[0]==="." || words.length === 0){
                     this.props.openApp("vscode");
                 } else {
-                    result = "Command '" + main + "' not found, or not yet implemented.<br>Available Commands: [ cd, ls, echo, clear, exit, mkdir,code]"; 
+                    result = "Command '" + main + "' not found, or not yet implemented.<br>Available Commands:[ cd, ls, pwd, echo, clear, exit, mkdir, openAllapps- code, spotify, chrome, about-vivek, todoist, trash, settings, sendmsg]";
```
Comment (vivek9patel) [inline src/components/apps/terminal.js:248]: just remove 'openAllapps- ' text, everything else is fine

File: src/components/apps/terminal.js [hunk 2]
```diff
@@ -263,7 +323,7 @@ export class Terminal extends Component {
                 result = "<img class=' w-2/5' src='./images/memes/used-sudo-command.jpg' />";
                 break;
             default:
-                result = "Command '" + main + "' not found, or not yet implemented.<br>Available Commands: [ cd, ls, echo, clear, exit, mkdir,code]";
+                result = "Command '" + main + "' not found, or not yet implemented.<br>Available Commands: [ cd, ls, pwd, echo, clear, exit, mkdir, openAllapps- code, spotify, chrome, about-vivek, todoist, trash, settings, sendmsg ]";
```
Comment (vivek9patel) [inline src/components/apps/terminal.js:326]: here also remove 'openAllapps- '

File: src/components/screen/all-applications.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ export class AllApplications extends React.Component {
         let apps = this.state.category === 0 ? [...this.state.apps] : getFrequentApps();
         apps.forEach((app, index) => {
             const props = {
-                name: app.title,
+                name: app.title.slice(0, 10) + (app.title.length > 10 ? "..." : ""),
```
Comment (vivek9patel) [inline src/components/screen/all-applications.js:50]: use CSS for slicing the text.

File: src/components/screen/desktop.js [hunk 4]
```diff
@@ -237,7 +238,7 @@ export class Desktop extends Component {
             if (this.state.desktop_apps.includes(app.id)) {
 
                 const props = {
-                    name: app.title,
+                    name: app.title.slice(0, 10) + (app.title.length > 10 ? "..." : ""),
```
Comment (vivek9patel) [inline src/components/screen/desktop.js:241]: use CSS

File: src/apps.config.js [hunk 5]
```diff
@@ -27,6 +28,15 @@ const apps = [
         desktop_shortcut: false,
         screen: displayTodoist,
     },
+    {
+        id: "calc",
+        title: "Calc",
```
Comment (vivek9patel) [inline src/apps.config.js:33]: change new app's name from 'calc' to 'calculator'

File: src/components/base/calcWindow.js [hunk 6]
```diff
@@ -0,0 +1,296 @@
+import React, { Component } from 'react';
```
Comment (vivek9patel) [inline src/components/base/calcWindow.js:1]: remove this whole file , we don't need this

File: src/components/screen/desktop.js [hunk 7]
```diff
@@ -3,6 +3,7 @@ import BackgroundImage from '../util components/background-image';
 import SideBar from './side_bar';
 import apps from '../../apps.config';
 import Window from '../base/window';
+import calcWindow from '../base/calcWindow';
```
Comment (vivek9patel) [inline src/components/screen/desktop.js:6]: remove this line also

File: src/components/screen/desktop.js [hunk 8]
```diff
@@ -279,6 +280,34 @@ export class Desktop extends Component {
         });
         return windowsJsx;
     }
+    renderCalcWindows = () => {
```
Comment (vivek9patel) [inline src/components/screen/desktop.js:283]: remove this function

File: src/components/screen/desktop.js [hunk 9]
```diff
@@ -502,6 +531,11 @@ export class Desktop extends Component {
                     {this.renderWindows()}
                 </div>
 
+                {/* Window Area */}
+                <div className="absolute h-full w-full bg-transparent" data-context="desktop-area">
+                    {this.renderCalcWindows()}
+                </div>
+
```
Comment (vivek9patel) [inline src/components/screen/desktop.js:538]: remove this also,
we have already implemented that, so no need to re-render all apps again.

**Comment by sunitapt** [comment]
> Great work there, especially in calculator app!
> I have few small changes.

Hey, Thanks! 
Can you plz tell me What are those changes?

**Review by vivek9patel — COMMENTED** [review]
> > Great work there, especially in calculator app!
> > I have few small changes.
> 
> Hey, Thanks!
> Can you plz tell me What are those changes?

Have already submitted review for those changes
you can find them in this page or in Files Changed section!

File: src/apps.config.js (hunk 5 above)
Comment (sunitapt) [reply src/apps.config.js:33]: hey! , in ubuntu, the name is "Calc" so should I change it or not?

Comment (vivek9patel) [reply src/apps.config.js:33]: well then it's fine, leave it

Comment (sunitapt) [reply src/apps.config.js:33]: Thanks!

File: src/components/apps/calc.js [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+        return (
+            <div className="h-full w-full bg-ub-drk-abrgn text-ubt-grey opacity-100 p-1 float-left font-normal">
+                <div>C-style arbitary precision calculator (version 2.12.7.2)</div>
+                <div>Calc is open software.</div>
+                <div>[ type "exit" to exit, "clear" to clear. ]</div>
+            <div className="text-white text-sm font-bold" id="calculator-body">
+                {this.state.terminal}
+            </div>
+            </div>
+        )
```
[the remark this answers is not in the document]
Comment (sunitapt) [reply src/components/apps/calc.js]: Added help command in calc. plz, check now! Let me know if there are any changes to that.

File: src/components/apps/terminal.js (hunk 1 above)
Comment (sunitapt) [reply src/components/apps/terminal.js]: done!

File: src/components/screen/all-applications.js (hunk 3 above)
Comment (sunitapt) [reply src/components/screen/all-applications.js]: yes, done!

**Comment by sunitapt** [comment]
hey @vivek9patel !
I have updated all recommended changes. 
Can you plz check them? also
let me know if there are any changes!

**Comment by vivek9patel** [comment]
> hey @vivek9patel !
> I have updated all recommended changes.
> Can you plz check them? also
> let me know if there are any changes!

LGTM!

**Comment by vivek9patel** [comment]
Have one question though
how does command works?
Screenshot 2021-09-01 at 4 43 09 PM

**Comment by sunitapt** [comment]
> Have one question though
> how does this command works?
> Screenshot 2021-09-01 at 4 43 09 PM

add values in sq braces like this 
ceil[2]

**Comment by vivek9patel** [comment]
> > Have one question though
> > how does this command works?
> > Screenshot 2021-09-01 at 4 43 09 PM
> 
> add values in sq braces like this
> ceil[2]

Got it!

- Just add that this commands example with square brackets
Screenshot 2021-09-01 at 4 50 05 PM

- increase the text width little bit so that chrome and message can be visible
Screenshot 2021-09-01 at 4 47 11 PM


and I'll merge it ✅

**Comment by sunitapt** [comment]
> > > Have one question though
> > > how does this command works?
> > > Screenshot 2021-09-01 at 4 43 09 PM
> > 
> > 
> > add values in sq braces like this
> > ceil[2]
> 
> Got it!
> 
> * Just add that this commands example with square brackets
> 
> Screenshot 2021-09-01 at 4 50 05 PM
> 
> * increase the text width little bit so that chrome and message can be visible
> 
> Screenshot 2021-09-01 at 4 47 11 PM
> 
> and I'll merge it

ok ill add square brackets. 
below Screenshot from the Linux desktop.  
send a message snapshot

**Comment by vivek9patel** [comment]
> > > > Have one question though
> > > > how does this command works?
> > > > Screenshot 2021-09-01 at 4 43 09 PM
> > > 
> > > 
> > > add values in sq braces like this
> > > ceil[2]
> > 
> > 
> > Got it!
> > 
> > * Just add that this commands example with square brackets
> > 
> > Screenshot 2021-09-01 at 4 50 05 PM
> > 
> > * increase the text width little bit so that chrome and message can be visible
> > 
> > Screenshot 2021-09-01 at 4 47 11 PM
> > and I'll merge it
> 
> ok ill add square brackets.
> below Screenshot from the Linux desktop.
> send a message snapshot

✅

**Comment by sunitapt** [comment]
> > > > > Have one question though
> > > > > how does this command works?
> > > > > Screenshot 2021-09-01 at 4 43 09 PM
> > > > 
> > > > 
> > > > add values in sq braces like this
> > > > ceil[2]
> > > 
> > > 
> > > Got it!
> > > 
> > > * Just add that this commands example with square brackets
> > > 
> > > Screenshot 2021-09-01 at 4 50 05 PM
> > > 
> > > * increase the text width little bit so that chrome and message can be visible
> > > 
> > > Screenshot 2021-09-01 at 4 47 11 PM
> > > and I'll merge it
> > 
> > 
> > ok ill add square brackets.
> > below Screenshot from the Linux desktop.
> > send a message snapshot

so should I change the width?

**Comment by sunitapt** [comment]
> > > > > > Have one question though
> > > > > > how does this command works?
> > > > > > Screenshot 2021-09-01 at 4 43 09 PM
> > > > > 
> > > > > 
> > > > > add values in sq braces like this
> > > > > ceil[2]
> > > > 
> > > > 
> > > > Got it!
> > > > 
> > > > * Just add that this commands example with square brackets
> > > > 
> > > > Screenshot 2021-09-01 at 4 50 05 PM
> > > > 
> > > > * increase the text width little bit so that chrome and message can be visible
> > > > 
> > > > Screenshot 2021-09-01 at 4 47 11 PM
> > > > and I'll merge it
> > > 
> > > 
> > > ok ill add square brackets.
> > > below Screenshot from the Linux desktop.
> > > send a message snapshot
> 
> so should I change the width? we can add a 'contact' as a title instead of sending a message. That will be directly visible

**Comment by vivek9patel** [comment]
> > > > > > Have one question though
> > > > > > how does this command works?
> > > > > > Screenshot 2021-09-01 at 4 43 09 PM
> > > > > 
> > > > > 
> > > > > add values in sq braces like this
> > > > > ceil[2]
> > > > 
> > > > 
> > > > Got it!
> > > > 
> > > > * Just add that this commands example with square brackets
> > > > 
> > > > Screenshot 2021-09-01 at 4 50 05 PM
> > > > 
> > > > * increase the text width little bit so that chrome and message can be visible
> > > > 
> > > > Screenshot 2021-09-01 at 4 47 11 PM
> > > > and I'll merge it
> > > 
> > > 
> > > ok ill add square brackets.
> > > below Screenshot from the Linux desktop.
> > > send a message snapshot
> 
> so should I change the width?

yes please
there are not much  app in desktop so I think at least in desktop name should be visible

**Comment by vivek9patel** [comment]
> > > > > > Have one question though
> > > > > > how does this command works?
> > > > > > Screenshot 2021-09-01 at 4 43 09 PM
> > > > > 
> > > > > 
> > > > > add values in sq braces like this
> > > > > ceil[2]
> > > > 
> > > > 
> > > > Got it!
> > > > 
> > > > * Just add that this commands example with square brackets
> > > > 
> > > > Screenshot 2021-09-01 at 4 50 05 PM
> > > > 
> > > > * increase the text width little bit so that chrome and message can be visible
> > > > 
> > > > Screenshot 2021-09-01 at 4 47 11 PM
> > > > and I'll merge it
> > > 
> > > 
> > > ok ill add square brackets.
> > > below Screenshot from the Linux desktop.
> > > send a message snapshot
> 
> so should I change the width? we can add a 'contact' as a title instead of sending a message. That will be directly visible

sure go ahed!

**Comment by sunitapt** [comment]
hey @vivek9patel ! 
done with changes. 
let me know if you want to change anything!
Thank you!

**Comment by vivek9patel** [comment]
Great job 👍

**Comment by sunitapt** [comment]
> Great job

Thank you so much for your support and guidance! It was nice working with you!

**Pull request merged**